tv_apps/smart-home/test/unit/home_test.js

```diff
@@ -0,0 +1,382 @@
+      assert.isTrue(testCase1.calledOnce);
+      assert.isFalse(testCase2.calledOnce);
+      subject.onCardInserted(subject.cardScrollable, card2, 1);
+      assert.isTrue(testCase2.calledOnce);
+    });
+  });
+
+  suite('onCardRemoved >', function() {
+    test('should remove specified cards from scrollable', function() {
+      var stub = this.sinon.stub();
+      subject.cardScrollable.removeNodes = stub;
+      subject.onCardRemoved(subject.cardScrollable, [0, 1]);
+      assert.isTrue(stub.withArgs([0, 1]).calledOnce);
+    });
+  });
+
+  suite('onMove >', function() {
+    test('should ask spatial navigator for next focus on non-scrollable target',
+    function() {
+      subject.spatialNavigator.m_focusedElement = document.createElement('div');
+      var stub = this.sinon.stub();
+      subject.spatialNavigator.move = stub;
+      subject.onMove('left');
+      assert.isTrue(stub.calledWith('left'));
+    });
+
+    test('should ask scrollable to move its focus on scrollable target',
+    function() {
+      var scrollable = new XScrollable();
+      scrollable.move = this.sinon.stub();
+      scrollable.move.returns(true);
+      subject.spatialNavigator.m_focusedElement = scrollable;
+      subject.onMove('right');
+      assert.isTrue(scrollable.move.calledWith('right'));
+    });
+
+    test('should ask scrollable to move its focus on scrollable target, but ' +
+    'fall-back to spatial navigator when scrollable didn\'t handle it',
+    function() {
+      var scrollable = new XScrollable();
+      scrollable.move = this.sinon.stub();
+      scrollable.move.returns(false);
+      subject.spatialNavigator.m_focusedElement = scrollable;
+      var spatialNavigatorMoveStub = this.sinon.stub();
+      subject.spatialNavigator.move = spatialNavigatorMoveStub;
+      subject.onMove('up');
+      assert.isTrue(scrollable.move.calledWith('up'));
+      assert.isTrue(spatialNavigatorMoveStub.calledWith('up'));
+    });
+  });
+
+  suite('onEnter >', function() {
+    test('should open settings on settings button', function() {
+      subject.handleFocus(settingsButton);
+
+      assert.equal(MozActivity.calls.length, 0);
+      subject.onEnter();
+      assert.equal(MozActivity.calls[0].name, 'configure');
+    });
+
+    test('should toggle edit mode on edit button', function() {
+      subject.handleFocus(editButton);
+
+      subject.edit.toggleEditMode = this.sinon.stub();
+      subject.onEnter();
+      assert.isTrue(subject.edit.toggleEditMode.calledOnce);
+    });
+
+    test('should launch application on card ', function() {
+      var testCard = document.createElement('div');
+      testCard.dataset.cardId = '123';
+
+      var testApp = new Application({});
+      testApp.launch = this.sinon.stub();
+      subject.cardManager.m_card['123'] = testApp;
+
+      subject.handleFocus(testCard);
+      subject.onEnter();
+      assert.isTrue(testApp.launch.calledOnce);
+    });
```

Comment:
I think we should also test `onEnter` when we are in edit mode.
That is to say, to examine if we bypass logic of `onEnter` or not when `this.edit.onEnter()` returns true.
However, I can accept that we add this test in next patch of unit test.